.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,92 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+     - v**
+  pull_request:
+    branches: [ master ]
+     
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
Building on a moving target could break the build, even if we change nothing.

I'd rather target a smaller, stable image. For example `debian:11`.

What do you think?

Reply:
This is an interesting one. I’m a hard proponent of always pinning versions, yet GitHub actions runs-on stanza might be worth an exception for the following reasons:

1. All actions documentation seems to recommend using either ‘ubuntu-latest’, ‘windows-latest’ or macos-latest’. I’ve never seen an exception to this outside of some nuanced osx builds targeting old versions. Here are the docs: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
2. the runs on stanza isn’t referencing the official ubuntu container like you  might expect. It is actually a blessed image from GitHub. See this quote 

> Note: The -latest runner images are the latest stable images that GitHub provides, and might not be the most recent version of the operating system available from the operating system vendor.

3. Anecdotally I’ve been using actions since 2018 and I’ve never seen a build break from ubuntu-latest.

Reply:
likewise, I've been using ubuntu-latest since the dawn of actions and haven't had any issues from it. Here's the repo that covers the runner images.
https://github.com/actions/runner-images

latest is currently ubuntu-22.04. For linux runners, we have a choice between latest (2204), 2204 pinned, and 2004

Reply:
That's all reasonable - let's stick with with `ubuntu-latest`. Thanks for the explanation!